src/main/java/com/maddyhome/idea/vim/extension/searchindex/VimSearchIndexExtension.kt

```diff
@@ -0,0 +1,183 @@
+import com.maddyhome.idea.vim.extension.VimExtension
+import com.maddyhome.idea.vim.extension.VimExtensionFacade
+import com.maddyhome.idea.vim.newapi.IjVimSearchGroup
+import com.maddyhome.idea.vim.newapi.ij
+import com.maddyhome.idea.vim.newapi.vim
+import com.maddyhome.idea.vim.state.mode.Mode
+import com.maddyhome.idea.vim.ui.ex.ExEntryPanel
+import java.awt.Component
+import java.awt.event.MouseEvent
+import javax.swing.KeyStroke
+
+internal object SearchIndex {
+  internal const val ID = "IdeaVimSearchIndex"
+  internal const val DISPLAY_NAME = "IdeaVim searchindex"
+  private var currentText = ""
+
+  fun update(text: String) {
+    currentText = text
+    val windowManager = WindowManager.getInstance()
+    ProjectManager.getInstance().openProjects.forEach {
+      val statusBar = windowManager.getStatusBar(it)
+      statusBar.updateWidget(ID)
+    }
+  }
+
+  fun getCurrentText(): String = currentText
+}
+
+internal class SearchIndexWidgetFactory : StatusBarWidgetFactory {
+  override fun getId() = SearchIndex.ID
+  override fun getDisplayName() = SearchIndex.DISPLAY_NAME
+  override fun createWidget(project: Project) = SearchIndexWidget(project)
+  override fun disposeWidget(widget: StatusBarWidget) {}
+  override fun canBeEnabledOn(statusBar: StatusBar) = true
+  override fun isAvailable(project: Project) = true
+}
+
+internal class SearchIndexWidget(project: Project) :
+  EditorBasedWidget(project),
+  StatusBarWidget.TextPresentation {
+
+  override fun ID() = SearchIndex.ID
+  override fun getPresentation() = this
+  override fun getClickConsumer(): Consumer<MouseEvent>? = null
+  override fun getText() = SearchIndex.getCurrentText()
+  override fun getTooltipText() = null
+  override fun getAlignment() = Component.RIGHT_ALIGNMENT
+}
+
+private class SearchAgainHandler(private val keys: List<KeyStroke>) : ExtensionHandler {
+  override fun execute(editor: VimEditor, context: ExecutionContext, operatorArguments: OperatorArguments) {
+    VimExtensionFacade.executeNormalWithoutMapping(keys, editor.ij)
+    VimSearchIndexExtension.updateSearchIndex(editor.ij, VimPlugin.getSearch())
+  }
+}
+
+internal class VimSearchIndexExtension : VimExtension, ModeChangeListener {
+  init {
+    LOG.info("Looking at SearchGroup implementation: ${VimPlugin.getSearch().javaClass}")
+  }
+
+  companion object {
+    private val LOG = logger<VimSearchIndexExtension>()
+
+    fun updateSearchIndex(editor: Editor, searchGroup: IjVimSearchGroup) {
+      if (searchGroup.lastSearchPattern.isNullOrEmpty()) {
+        return
+      }
+      val pattern = searchGroup.lastSearchPattern
+      if (pattern.isNullOrEmpty()) {
+        return
+      }
+
+      LOG.info("search index - Processing pattern: $pattern")
+
+      // FIXME: global search option -> previous search option
+      val ignoreCase = injector.options(editor = editor.vim).ignorecase
+      LOG.info("Ignore case: $ignoreCase")
+
+      val results = injector.searchHelper.findAll(
```

Comment:
Searching twice on each `n` key looks like a performance issue. I left a comment about the new search listener. I think, it may also send the information about the number of found results.